apps/application-system/api/src/app/modules/application/application.controller.ts

```diff
@@ -72,4 +78,37 @@ export class ApplicationController {
 
     return updatedApplication
   }
+
+  @Put(':id/externalData')
+  @ApiOkResponse({ type: Application })
+  async updateExternalData(
+    @Param('id') id: string,
+    @Body()
+    externalDataDto: PopulateExternalDataDto,
+  ): Promise<Application> {
+    // TODO how can we know if the requested data-providers are actually associated with this given form?
+    const application = await this.applicationService.findById(id)
+
+    if (!application) {
```

Comment:
`await this.applicationService.findById(id)` 
will return null if the application is not found then I think this check does not work

Reply:
!null is truthy, so it should work splendidly